src/core/core.datasetController.js

```diff
@@ -1,52 +1,10 @@
 'use strict';
 
-var helpers = require('../helpers/index');
-
-var resolve = helpers.options.resolve;
-
-var arrayEvents = ['push', 'pop', 'shift', 'splice', 'unshift'];
-
-/**
- * Hooks the array methods that add or remove values ('push', pop', 'shift', 'splice',
- * 'unshift') and notify the listener AFTER the array has been altered. Listeners are
- * called on the 'onData*' callbacks (e.g. onDataPush, etc.) with same arguments.
- */
-function listenArrayEvents(array, listener) {
-	if (array._chartjs) {
-		array._chartjs.listeners.push(listener);
-		return;
-	}
-
-	Object.defineProperty(array, '_chartjs', {
-		configurable: true,
-		enumerable: false,
-		value: {
-			listeners: [listener]
-		}
-	});
-
-	arrayEvents.forEach(function(key) {
-		var method = 'onData' + key.charAt(0).toUpperCase() + key.slice(1);
-		var base = array[key];
-
-		Object.defineProperty(array, key, {
-			configurable: true,
-			enumerable: false,
-			value: function() {
-				var args = Array.prototype.slice.call(arguments);
-				var res = base.apply(this, args);
-
-				helpers.each(array._chartjs.listeners, function(object) {
-					if (typeof object[method] === 'function') {
-						object[method].apply(object, args);
-					}
-				});
-
-				return res;
-			}
-		});
-	});
-}
+import {arrayEquals, extend, inherits, isArray, isObject, merge, _merger, noop, valueOrDefault} from '../helpers/helpers.core';
```

Comment:
Maybe `arrayEquals` would belong in `helpers.collection`?

Reply:
Year, `isArray` and `each` maybe too. This is not complete, just a start.

Reply:
I'd probably leave `isArray` in `core` with `isObject`. I'd hope that we can just get rid of `each` as it doesn't really seem necessary to me most of the time (e.g. we can just use the built in `forEach` instead)